plugins/base/src/main/kotlin/renderers/html/HtmlRenderer.kt

```diff
@@ -66,20 +73,15 @@ open class HtmlRenderer(
         val additionalClasses = node.style.joinToString(" ") { it.toString().toLowerCase() }
         return when {
             node.hasStyle(ContentStyle.TabbedContent) -> div(additionalClasses) {
-                val secondLevel = node.children.filterIsInstance<ContentComposite>().flatMap { it.children }
-                    .filterIsInstance<ContentHeader>().flatMap { it.children }.filterIsInstance<ContentText>()
-                val firstLevel = node.children.filterIsInstance<ContentHeader>().flatMap { it.children }
-                    .filterIsInstance<ContentText>()
-
-                val renderable = sortTabs(tabSortingStrategy, firstLevel.union(secondLevel))
+                val contentTabs= node.extra[ContentTabsExtra]?.tabs ?: throw IllegalStateException("The tabbed content must have ContentTabsExtra")
 
                 div(classes = "tabs-section") {
                     attributes["tabs-section"] = "tabs-section"
-                    renderable.forEachIndexed { index, node ->
+                    contentTabs.forEachIndexed { index, node ->
                         button(classes = "section-tab") {
                             if (index == 0) attributes["data-active"] = ""
-                            attributes["data-togglable"] = node.text
-                            text(node.text)
+                            attributes["data-togglable"] = node.toggleableContentTypes.joinToString(",")
```

Comment:
Yes, I don't see any contradiction with my comment - it actually outlines the same problem here.

> What do you propose?

Extract the mapping logic into a separate function, like `fun ToggleableContentType.toTogglableAttributeValue(): String`, so that the attribute value and the name of the enum entry are not coupled and can be changed individually without breaking everything.

Reply:
> Extract the mapping logic into a separate function, like fun ToggleableContentType.toTogglableAttributeValue(): String

So how plugin authors can add their own `ContentType` (tabs)?

Reply:
Good question. There could be an extension point for configuring and styling tabs for the HTML format. The function itself could be open until proper API is introduced. Even not providing the ability to add their own tabs, in my opinion, is better than relying on `interface.toString()`, as it can be virtually anything and might lead to the same problem as in `#2803`

Reply:
`interface.toString()` is not  necessary. It can be the getter `.name``

Reply:
@IgnatBeresnev  WDYT?